frontend/website/src/components/login.svelte

```diff
@@ -0,0 +1,601 @@
+<script lang="ts">
+	import { onMount } from "svelte";
+	import { Turnstile } from "svelte-turnstile";
+	import { loginMode, sessionToken } from "../store/login";
+	import { focusTrap } from "../lib/focusTrap";
+	import LoginField, { newField } from "./loginField.svelte";
+	import { z } from "zod";
+	import { getContextClient } from "@urql/svelte";
+	import { graphql } from "../gql";
+
+	const client = getContextClient();
+
+	let show = false;
+
+	let globalMessage = "";
+	let globalIsError = false;
+
+	function closeLogin() {
+		show = false;
+		setTimeout(() => {
+			loginMode.set(0);
+		}, 100);
+	}
+
+	function handleOnMouseDown(event: MouseEvent) {
+		if (event.target === event.currentTarget) {
+			closeLogin();
+		}
+	}
+
+	function handleKeyDown(event: KeyboardEvent) {
+		if (event.key === "Escape") {
+			closeLogin();
+		}
+	}
+
+	onMount(() => {
+		document.body.classList.add("no-overflow");
+		setTimeout(() => {
+			show = true;
+		}, 100);
+
+		return () => {
+			document.body.classList.remove("no-overflow");
+		};
+	});
+
+	let username = newField({
+		id: "username",
+		type: "text",
+		label: "Username",
+		placeholder: "Username",
```

Comment:
I dont like how placeholder is always the same as label maybe we can put some example as placeholder or make it empty?